### DISCUSSION TITLE
ValidatingRequest?

### DISCUSSION THREAD

**Discussion by javz3** [SixLabors/ImageSharp.Web#179] [Q&A] [answered]
Hi everyone, 

I was wondering if there is an ImageSharp equivalent for ImageProcessors ValidatingRequest which allows devs to add custom instructions based on the URL request (Please check https://imageprocessor.org/imageprocessor-web/imageprocessingmodule/ - ImageProcessingModule.ValidatingRequest) 

I want to throw a custom error/message when the requested URL contains incorrect resolutions, so people don't overload the server with numerous images - only a predefined list of resolutions will be generated and accessible.

The reason why I ask is Umbraco V9 uses ImageSharp where it once used ImageProcessor. I can't seem to find anything on the documentation for ImageSharp - https://docs.sixlabors.com/articles/imagesharp.web/gettingstarted.html

Thanks

**Comment by JimBobSquarePants** [comment, marked as answer]
There’s actually full API docs at the same location. You’re looking for [`ImageSharpMiddlewareOptions.OnParaeCommandsAsync`](https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddlewareOptions.html#SixLabors_ImageSharp_Web_Middleware_ImageSharpMiddlewareOptions_OnParseCommandsAsync).

**Reply by javz3** [reply]
Thanks for the quick response @JimBobSquarePants is there an example to see how this functions? Thanks again!

**Reply by JimBobSquarePants** [reply]
Sure... Here's the default implementation
https://github.com/SixLabors/ImageSharp.Web/blob/52bcb16fee6d600228769c459d961018ac9c2baa/src/ImageSharp.Web/Middleware/ImageSharpMiddlewareOptions.cs#L20-L44
